packages/openapi-typescript/src/types.ts

```diff
@@ -447,18 +447,21 @@ export type SchemaObject = {
 
 export interface StringSubtype {
   type: "string";
+  enum?: (string | ReferenceObject)[];
```

Comment:
I’m not completely happy with the `enum` type changes; we may need to change these in the future. But it just made this change easier.